packages/dds/tree/src/feature-libraries/node-key/nodeKeyManager.ts

```diff
@@ -92,13 +109,26 @@ class MockNodeKeyManager implements NodeKeyManager {
 	}
 
 	public localizeNodeKey(key: StableNodeKey): LocalNodeKey {
-		return brand(Number.parseInt(key.substring(30), 16) as SessionSpaceCompressedId);
+		const localNodeKey = this.tryLocalizeNodeKey(key);
+		assert(localNodeKey !== undefined, "Key is not compressible");
+		return localNodeKey;
 	}
 
 	public stabilizeNodeKey(key: LocalNodeKey): StableNodeKey {
 		return brand(this.createMockStableId(extractFromOpaque(key)));
 	}
 
+	public tryLocalizeNodeKey(key: string): LocalNodeKey | undefined {
+		const localNodeKey = Number.parseInt(key.substring(24), 16);
+		if (Number.isNaN(localNodeKey)) {
```

Comment:
You can use `isStableId` to validate that `key` is a StableNodeKey, and if so you can assume that the parseInt will succeed - no need for the isNaN check.

Reply:
Or better, I would consider making an `isStableNodeKey()` type guard (which just calls `isStableId` inside) for maximum type safety.

Reply:
Added `isStableNodeKey` type guard function to `nodeKeyManager` and `nodeKeys` and removed `isNaN` check